packages/alfa-rules/src/common/expectation/get-colors.ts

```diff
@@ -0,0 +1,227 @@
+import { Element } from "@siteimprove/alfa-dom";
+import { Device } from "@siteimprove/alfa-device";
+import { None, Option } from "@siteimprove/alfa-option";
+import { Current, Percentage, RGB, System } from "@siteimprove/alfa-css";
+import { Style } from "@siteimprove/alfa-style";
+import { Iterable } from "@siteimprove/alfa-iterable";
+
+const { map } = Iterable;
+
+/**
+ * Determine the approximate foreground color of an element if possible.
+ */
+export function getForeground(
```

Comment:
I decided against splitting `getFG` and `getBG` in separate files since they both depend on the same bits that would then need to be shared…